tests/serverpod_test_server/test_integration/test_tools/database_operations_test.dart

```diff
@@ -416,4 +416,99 @@ void main() {
       rollbackDatabase: RollbackDatabase.disabled,
     );
   });
+
+  withServerpod(
+    'Given rollbackDatabase is not disabled ',
+    rollbackDatabase: RollbackDatabase.afterEach,
+    (sessionBuilder, _) {
+      var session = sessionBuilder.build();
+
+      group('when creating UniqueData with the same unique value', () {
+        late Future failingInsert;
+        setUp(() async {
+          await UniqueData.db.insertRow(
+            session,
+            UniqueData(email: 'test@test.com', number: 1),
+          );
+          failingInsert = UniqueData.db.insertRow(
+            session,
+            UniqueData(email: 'test@test.com', number: 1),
+          );
+        });
+
+        test('then should throw database exception', () async {
+          await expectLater(
+            failingInsert,
+            throwsA(
+              allOf(
+                isA<DatabaseException>(),
+              ),
+            ),
+          );
+        });
+
+        test(
+            'then catching database exception should not prevent further database operations',
+            () async {
+          SimpleData? simpleData;
+          try {
+            await failingInsert;
+          } on DatabaseException catch (_) {
+            simpleData =
+                await SimpleData.db.insertRow(session, SimpleData(num: 123));
+          }
+
+          expect(simpleData, isNotNull);
+          expect(simpleData?.num, 123);
+        });
+      });
+
+      test(
```

Comment:
Question: Are we guaranteed that the first transaction will always happen first? Since they are happening in "parallel" we shouldn't know or care which of the inserts happen first right?

Reply:
You're right! The test description should be updated to reflect that.

Reply:
4bbe168